### PR TITLE
6 routing

### PR DESCRIPTION
**Pull request by mateuszkrzak**

### CODE REVIEW

File: app/src/common/components/MovieList/MovieList.jsx [hunk 1]
```diff
@@ -21,7 +22,11 @@ export default class MovieList extends React.Component {
 
     return (
       <div data-qa="movie-list" styleName="list">
-        {this.props.movies.map(movie => <MovieElement key={movie.id} {...movie} />)}
+        {this.props.movies.map(movie => (
+          <Link to={`/film/${movie.id}`} key={movie.id} href={`/film/${movie.id}`}>
```
[the remark this answers is not in the document]
Comment (mateuszkrzak) [reply app/src/common/components/MovieList/MovieList.jsx]: I removed it and disabled 'jsx-a11y/anchor-is-valid' rule in eslint.

File: app/src/containers/MovieDetailsPage/MovieDetailsPage.jsx [hunk 2]
```diff
@@ -11,7 +12,16 @@ import getMovieByIdAndSimilarMoviesAsync from './actions';
...
+    this.props.getMovieByIdAndSimilarMoviesAsync(this.props.match.params.id);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.match.params.id !== prevProps.match.params.id) {
+      this.props.getMovieByIdAndSimilarMoviesAsync(this.props.match.params.id);
+      window.scrollTo(0, 0);
+    }
+
+    return this.props.match.params.id !== prevProps.match.params.id;
```
[the remark this answers is not in the document]
Comment (mateuszkrzak) [reply app/src/containers/MovieDetailsPage/MovieDetailsPage.jsx]: Right, missed this after refactoring from shouldComponentUpdate.

File: app/src/bootstrap.jsx [hunk 3]
```diff
@@ -1,16 +1,29 @@
...
 
 render(
   <Provider store={store}>
-    <App />
+    <Router>
+      <App>
+        <Switch>
+          <Route path="/film/:id" component={MovieDetailsPage} />
+          <Route path="/search/:query" component={SearchPage} />          
+          <Route path="/" component={SearchPage} />
```
[the remark this answers is not in the document]
Comment (mateuszkrzak) [reply app/src/bootstrap.jsx]: It seems to working without `exact`, can you provide reasoning for this?

File: app/src/bootstrap.jsx [hunk 4]
```diff
@@ -1,16 +1,29 @@
...
 import App from './App';
 import store from './redux/store';
+import SearchPage from './containers/SearchPage/SearchPage';
+import MovieDetailsPage from './containers/MovieDetailsPage/MovieDetailsPage';
+import NotFoundPage from './containers/NotFoundPage/NotFoundPage';
 
 render(
   <Provider store={store}>
-    <App />
+    <Router>
```
[the remark this answers is not in the document]
Comment (mateuszkrzak) [reply app/src/bootstrap.jsx]: Ok.

File: app/src/bootstrap.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (olvnikon) [reply app/src/bootstrap.jsx]: image

[the remark this answers is not in the document]
Comment (mateuszkrzak) [reply app/src/bootstrap.jsx]: Oh right, thanks!

**Pull request merged**